karapace/serialization.py

```diff
@@ -108,63 +109,69 @@ async def close(self):
 
 
 class SchemaRegistrySerializerDeserializer:
-    def __init__(self, **cfg):
-        if not cfg.get("config_path"):
-            raise ValueError("config_path is empty or missing")
-        config_path = cfg["config_path"]
+    def __init__(
+        self,
+        config_path: str,
+        name_strategy: str = "topic_name",
+        **cfg,  # pylint: disable=unused-argument
+    ) -> None:
         self.config = read_config(config_path)
         self.state_lock = asyncio.Lock()
         registry_url = f"http://{self.config['registry_host']}:{self.config['registry_port']}"
         registry_client = SchemaRegistryClient(registry_url)
-        self.subject_name_strategy = NAME_STRATEGIES[cfg.get("name_strategy", "topic_name")]
-        self.registry_client = registry_client
-        self.ids_to_schemas = {}
-        self.schemas_to_ids = {}
+        self.subject_name_strategy = NAME_STRATEGIES[name_strategy]
+        self.registry_client: Optional[SchemaRegistryClient] = registry_client
+        self.ids_to_schemas: Dict[int, TypedSchema] = {}
+        self.schemas_to_ids: Dict[str, int] = {}
 
-    async def close(self):
+    async def close(self) -> None:
         if self.registry_client:
             await self.registry_client.close()
             self.registry_client = None
 
     def get_subject_name(self, topic_name: str, schema: str, subject_type: str, schema_type: SchemaType) -> str:
-        schema = TypedSchema.parse(schema_type, schema)
+        schema_typed = TypedSchema.parse(schema_type, schema)
         namespace = "dummy"
         if schema_type is SchemaType.AVRO:
-            namespace = schema.schema.namespace
+            namespace = schema_typed.schema.namespace
         if schema_type is SchemaType.JSONSCHEMA:
-            namespace = schema.to_json().get("namespace", "dummy")
+            namespace = schema_typed.to_json().get("namespace", "dummy")
         return f"{self.subject_name_strategy(topic_name, namespace)}-{subject_type}"
 
     async def get_schema_for_subject(self, subject: str) -> TypedSchema:
+        assert self.registry_client, "must not after the object is closed."
```

Comment:
must not what?

Reply:
Ooops, my bad, updated it to `must not call this method after the object is closed.`